app/controllers/resources_controller.rb

```diff
@@ -13,7 +13,7 @@ def new
   end
 
   def show
-    redirect_to(params.merge(:action => "edit"))
+    redirect_to :action => :edit
```

Comment:
The reason to do `params.merge` was so that any additional parameters also get passed to the new URL.